### ISSUE TITLE
mapPlot() axis labels shown inconsistently

### ISSUE THREAD

**Issue by clayton33** [dankelley/oce#1349]
### Short summary of problem

`mapPlot()` axis labels are shown inconsistently. For me, this is occurring for latitude labels Happens for both interactive work , and saved output.

### What you did
```
library(oce)
library(ocedata)
data('coastlineWorldFine')
par(mar=c(3,3,1,1))
mapPlot(coastlineWorldFine, 
        longitudelim = c(-60, -58), 
        latitudelim = c(45.5, 46.5),
        proj = '+proj=merc',
        col = 'grey')
```

### What you expected to happen

Axis labels for both longitude and latitude.

### What happened

![mapplotissue](https://user-images.githubusercontent.com/3463161/33895233-6de69f8a-df36-11e7-80b8-0d73a123b1b0.png)
No labels for latitude. Examples of working mapPlot() labels have been sent privately in e-mail. 

### How urgent is this?

Not urgent

### Output from sessionInfo()

```
R version 3.4.1 (2017-06-30)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=English_Canada.1252  LC_CTYPE=English_Canada.1252    LC_MONETARY=English_Canada.1252
[4] LC_NUMERIC=C                    LC_TIME=English_Canada.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] marmap_0.9.6   ocedata_0.1.4  oce_0.9-23     gsw_1.0-5      testthat_1.0.2

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.14        raster_2.5-8        magrittr_1.5        bit_1.1-12          lattice_0.20-35    
 [6] R6_2.2.0            rlang_0.1.2         adehabitatMA_0.3.11 stringr_1.1.0       blob_1.1.0         
[11] plyr_1.8.4          tools_3.4.1         rgdal_1.2-5         grid_3.4.1          DBI_0.7            
[16] bit64_0.9-7         digest_0.6.11       tibble_1.3.4        crayon_1.3.2        reshape2_1.4.2     
[21] ncdf4_1.15          memoise_1.0.0       shape_1.4.3         RSQLite_2.0         sp_1.2-4           
[26] stringi_1.1.2       compiler_3.4.1  
```

**Comment by dankelley** [comment]
Thanks for the report, which actually follows our template, and makes the template seem useful :-)

I'll have a look this afternoon or possibly sometime tomorrow near noon.

**Comment by dankelley** [comment]
Test code added to oce-issues repo https://github.com/dankelley/oce-issues/tree/master/13xx/1349

**Comment by dankelley** [comment]
I see why this is happening. Axis labelling is tricky because it requires inverse projection. The reason is that the edges of the plot box have a complicated connection to the lon-lat values.  This particular case is pretty simple because the plot span is small, but consider the case of a larger domain, with curved lines of lon and lat, and imagine we are looking to label a latitude value of 50N (for example). What the code tries to do is to find a spot *on the edge of the plot box* that has this latitude, along with a longitude that is on the lon-lat function that describes the left-hand side of the box. This necessarily involves doing inverse map projections, and that's messy.

Looking at the code in `R/map.R`, I see that I have to methods. For the side=1 axis, I use `optimize` to find the longitude that, combined with lat=50N, yields a spot on the vertical line at `par(usr)[1]` in the map coordinates. However, for side=2, I use a different method, trying to bracket between a set of longitude values: -360, -340, ... 340, and 360. I check each of those values, to see if the left-hand side of the plot box hits. I think the problem is that this is too coarse, jumping by 20deg.

I'll call these the **optimize** and **bracket** methods.

I'm not completely sure why this side=2 method is not working, but I think it may have advantages in other cases, so I am not too keen on removing it! However, what I think will work fine is to try using **optimize** first, using its results if it works, and then to try **bracket** if we have to. All we are doing is trying to find an inverse here, and so I think it's fine to try the two methods in sequence.

PS. this is all made hard by the fact that some projections will have labellable (sp?) axes, and others won't. Indeed, some projections will have portions that can be labelled and portions that cannot (think of a global-scale view that has part of the earth outside the plot area ... we could get an axis labelling just for the portion where the land goes offscale.

**Comment by dankelley** [comment]
Oh, cancel those fine-sounding words. The problem was that in the **bracket** method I was using a `uniroot` call with a `tol` of 1degree. Sheesh -- that's 100 km in lat and even at "our" lat its a whack of kilometers.  When I change that to 0.01deg (at most then 1km scale) it all works fine. I'll clean up my debugging and commit within the hour, but in the meantime, the screenshot shows how silly I am in debugging statements.
<img width="926" alt="screen shot 2017-12-12 at 4 17 46 pm" src="https://user-images.githubusercontent.com/99469/33906427-465a70fa-df58-11e7-95a6-be09aaf6a6f7.png">

**Comment by dankelley** [comment]
I think this is fixed in "develop" commit bb901d61ee23b04683f468f3d5bc24ea43a9f66b. Click link to see code changes, the most important of which is the smaller value for the `tol` argument of a call to `uniroot`.

**Comment by clayton33** [comment]
Looks good ! Thanks !
